From 68a0412fd631595c54c26c27183e6434de195ba3 Mon Sep 17 00:00:00 2001
From: Luca Beltrame <lbeltrame@kde.org>
Date: Sat, 6 Aug 2022 15:39:15 +0200
Subject: [PATCH] Fix pimcommon CMake

The configuration file for pimcommon was actually missing KIOCore, and
at the same time it worked by accident by linking QtNetwork, which
wasn't a public link target.

This change adds the KIOCore dependency where it belongs and fixes the
link targets properly by adding the actual dependencies (Qt Core, Gui,
Widgets, DBus). Qt Network is a private link target, so it is moved
where it belongs.

Reviewed and ack'ed by krop on IRC.

(squashed with 37bd45ce626e4)
---
 src/pimcommon/CMakeLists.txt              | 7 +++++--
 src/pimcommon/KF5PimCommonConfig.cmake.in | 4 ++++
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/pimcommon/CMakeLists.txt b/src/pimcommon/CMakeLists.txt
index 7af8744..943094e 100644
--- a/src/pimcommon/CMakeLists.txt
+++ b/src/pimcommon/CMakeLists.txt
@@ -192,11 +192,14 @@ generate_export_header(KF5PimCommon BASE_NAME pimcommon)
 
 target_link_libraries(KF5PimCommon
     PUBLIC
-    Qt${QT_MAJOR_VERSION}::Network
+    Qt${QT_MAJOR_VERSION}::Gui
+    Qt${QT_MAJOR_VERSION}::Widgets
+    Qt${QT_MAJOR_VERSION}::DBus
     KF5::KIOCore
     KF5::ConfigWidgets
     KF5::PimTextEdit
     PRIVATE
+    Qt${QT_MAJOR_VERSION}::Network
     KF5::I18n
     KF5::Libkdepim
     KF5::Archive
@@ -225,7 +228,7 @@ set_target_properties(KF5PimCommon PROPERTIES
 
 install(TARGETS
     KF5PimCommon
-    EXPORT KF5PimCommonTargets ${KF5_INSTALL_TARGETS_DEFAULT_ARGS} 
+    EXPORT KF5PimCommonTargets ${KF5_INSTALL_TARGETS_DEFAULT_ARGS}
     )
 
 if (BUILD_TESTING)
diff --git a/src/pimcommon/KF5PimCommonConfig.cmake.in b/src/pimcommon/KF5PimCommonConfig.cmake.in
index f41bc6a..0a39f97 100644
--- a/src/pimcommon/KF5PimCommonConfig.cmake.in
+++ b/src/pimcommon/KF5PimCommonConfig.cmake.in
@@ -5,5 +5,9 @@
 include(CMakeFindDependencyMacro)
 find_dependency(KF5Config "@KF5_MIN_VERSION@")
 find_dependency(KF5PimTextEdit "@KPIMTEXTEDIT_LIB_VERSION@")
+find_dependency(KF5KIO "@KF5_MIN_VERSION@")
+find_dependency(Qt@QT_MAJOR_VERSION@Widgets "@QT_REQUIRED_VERSION@")
+find_dependency(Qt@QT_MAJOR_VERSION@DBus "@QT_REQUIRED_VERSION@")
+find_dependency(Qt@QT_MAJOR_VERSION@Gui "@QT_REQUIRED_VERSION@")
 
 include("${CMAKE_CURRENT_LIST_DIR}/KF5PimCommonTargets.cmake")
-- 
2.37.1

